CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Hmm, I'm not quite ready to drop support for Debian 9 and Ubuntu 18.04 yet...

Reply:
Looks like Debian 9 has a 3.16 backport, and there are plenty of reasonable ways to get newer cmake on Ubuntu 18.04, so I'm discarding this concern....